src/Components/NavbarComponent/NewPost.js

```diff
@@ -10,36 +26,53 @@ const InternalNewPost = props => {
             <div class='col-md-10 offset-md-1 col-xs-12'>
               <form>
                 <fieldset>
+                  
+                  {/* ---- Title ---- */}
                   <fieldset class='form-group'>
                     <input
                       type='text'
                       class='form-control form-control-lg'
+                      onChange={event => setTitle(event.target.value)}
                       placeholder='Article Title'></input>
                   </fieldset>
+
+                  {/* ---- Description ---- */}
                   <fieldset class='form-group'>
                     <input
                       type='text'
                       class='form-control'
+                      onChange={event => setDescription(event.target.value)}
                       placeholder="What's this article about?"></input>
                   </fieldset>
+
+                  {/* ---- Content ---- */}
                   <fieldset class='form-group'>
                     <textarea
                       class='form-control'
                       rows='8'
+                      onChange={event => setContent(event.target.value)}
                       placeholder='Write your article (in markdown)'></textarea>
                   </fieldset>
+
+                  {/* ---- Tags ---- */}
                   <fieldset class='form-group'>
                     <input
                       type='text'
                       class='form-control'
+                      onChange={event => setTags(event.target.value)}
                       placeholder='Enter tags'></input>
                     <div class='tag-list'></div>
                   </fieldset>
+
                   <button
                     class='btn btn-lg pull-xs-right btn-primary'
-                    type='button'>
+                    type='button'
+                    onClick={()=> {
+                      props.onPostArticleClicked(title, description, content, tags, props.currentSlug );
```

Comment:
After this POST api call, the content of the new article will be posted to server, and a article slug will be returned, then we redirect to the article detail component, use this new slug to load the article and the comments like the normal way.